### PR TITLE
feat: general letter + cover sheet for print and post (FPLA-1170/1086)

### PR DESCRIPTION
**Pull request by joez-f**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1170

### Change description ###

Added new service that creates a new docmosis document (general letter + cover sheet in one document) that will accompany the main document as part of print and post. Document contains representative address that needs to fit specifically within the envelope window, would most likely require real letters to be sent by Xerox for proper testing.

--NOTE: service not currently utilised until other printing stories are ready (FPLA-1096 and FPLA-1173). Will be tested as part of #826 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by joez-f** [comment]
[general_letter.pdf](https://github.com/hmcts/fpl-ccd-configuration/files/4133932/general_letter.pdf)
example general letter

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java [hunk 1]
```diff
@@ -278,7 +278,7 @@ private String formatEPODateTime(LocalDateTime dateTime) {
 
     private String getFormattedRemovalAddress(CaseData caseData) {
         if (caseData.getEpoRemovalAddress() != null) {
-            return caseData.getEpoRemovalAddress().getAddressAsString();
+            return caseData.getEpoRemovalAddress().getAddressAsString(", ");
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java]: Thoughts of replacing this method with an optional, `return Optional.ofNullable(caseData.getEpoRemovalAddress()).map(THIS_LINE).orElse("")`

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/DocmosisCoverDocumentsServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        return Representative.builder()
+            .fullName("Mark Jones")
+            .address(Address.builder()
+                .addressLine1("1 Petty France")
+                .addressLine2("St James's Park")
+                .postTown("London")
+                .build())
+            .build();
+    }
+
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/DocmosisCoverDocumentsServiceTest.java]: 👀

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java (hunk 1 above)
Comment (joez-f) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java]: 👍

**Comment by joez-f** [comment]
[cover_documents.pdf](https://github.com/hmcts/fpl-ccd-configuration/files/4159522/cover_documents.pdf)
Example of PDF with both documents

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.formatCCDCaseNumber;
+
+class CaseDetailsHelperTest {
```
Comment (akomisarek) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelperTest.java]: Being picky - tests for null / single element number? :)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.google.common.base.Splitter;
+
+public class CaseDetailsHelper {
+    private CaseDetailsHelper() {
+        // NO OP
+    }
+
+    public static String formatCCDCaseNumber(Long caseNumber) {
+        if (String.valueOf(caseNumber).length() != 16) {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java]: `String.valueOf(caseNumber)` this is being called twice, make it a variable

**Pull request merged**